modules/videoio/src/cap.cpp

```diff
@@ -628,4 +658,37 @@ int VideoWriter::fourcc(char c1, char c2, char c3, char c4)
     return (c1 & 255) + ((c2 & 255) << 8) + ((c3 & 255) << 16) + ((c4 & 255) << 24);
 }
 
+namespace detail {
+static void fillRequiredVideoWriterParametersWithDefaults(VideoWriterParameters& parameters)
```

Comment:
Not sure that this is correct approach.

Backends may have own specific parameters.
So one function is not enough.
Also it is not enough one such function per backend - for example, file container / codec tuning parameters.

So it is better to replace this by stuff by similar to Python `.get(key, default_value)` method:
```
int get(std::map<int, int>& map, int key, const int defaultValue);

// or more generic (not needed for now)
template<...>
getKeyOrDefault(std::map<TKey, TValue>& map, const TKey& key, const TValue defaultValue)
```

Reply:
The goal of this function is to be a configuration point for required *common* parameters, e.g. `isColor`, to share 1 default value for various backends.

I like your way of getting codec/backend specific parameters and will use it in further.

Maybe better name (e.g. `fillCommonRequiredWriterParameters`) should remove such confusion or it's better to remove it anyway?